### PR TITLE
[feature] Add presentation tier

### PR DESCRIPTION
**Pull request by jccolina**
Desriptin
=========
- Add presentation classes
- Modified services to add some required getters
- Modified other classes to add some validations

### CODE REVIEW

File: src/colinajose/data/ReadWriteCSV.java [hunk 1]
```diff
@@ -46,19 +46,23 @@ public boolean writeEntries(MyArrayList<MyHashMap<String, String>> entries) {
...
+                for (int i = 0; i < entries.size(); i++) {
+                    MyHashMap<String, String> entry = entries.get(i);
+                    String[] line = new String[headers.length];
+                    for (int j = 0; j < headers.length; j++) {
+                        line[j] = entry.get(headers[j]);
+                    }
+                    writer.writeNext(line);
                 }
-                writer.writeNext(line);
+                writer.close();
```
Comment (JurgenPadilla) [inline src/colinajose/data/ReadWriteCSV.java]: Why you are only closing the writer on one path and leaving it open on the other?

File: src/colinajose/service/SchoolService.java [hunk 2]
```diff
@@ -22,188 +23,265 @@
...
+    public Teacher getTeacher(String teacherId) {
         return SearchService.getTeacher(this.school, teacherId);
     }
-    public Student getStudent(String studentId){
+
+    public Student getStudent(String studentId) {
         return SearchService.getStudent(this.school, studentId);
     }
+
+    private <T extends BaseEntity> MyArrayList<MyHashMap<String, String>> elementsToEntries(MyCircularDoublyLinkedList<T> elements) {
```
Comment (JurgenPadilla) [inline src/colinajose/service/SchoolService.java]: Jose, I don't like this method you are not utilizing Map correctly. This method should return a Map not a list of Maps.
The returned Map keys will be the Ids and the values will be the actual object (T).
As this is you are tightly coupling the Services with the Presentation. Handling everything as a String is not ok.
If need to, please modify the MyHashMap class to include two methods.
public MyArrayList<T> getKeys();
public MyArrayList<U> getValues();

**Review by JurgenPadilla — CHANGES_REQUESTED** [review]

File: src/colinajose/service/SchoolService.java (hunk 2 above)
Comment (jccolina) [reply src/colinajose/service/SchoolService.java]: @JurgenPadilla fixed in next commit

File: src/colinajose/data/ReadWriteCSV.java (hunk 1 above)
Comment (jccolina) [reply src/colinajose/data/ReadWriteCSV.java]: @JurgenPadilla fixed in next commit

**Review by JurgenPadilla — APPROVED** [review]

**Pull request merged**